### ISSUE TITLE
Possible null pointer dereference

### ISSUE THREAD

**Issue by JTHartikainen** [TheFinnishSocialInsuranceInstitution/KantaCDA-API#6]
KantaCDA-API-master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/Kasaaja.java:225 Possible null pointer dereference of leimakentat in fi.kela.kanta.cda.Kasaaja.addIdFields(POCDMT000040ClinicalDocument, LeimakentatTO, String) 

There is an if block earlier in the method, checking if supplied instance of LeimakentatTO is null, but it is not checked later and could lead to NPE on row 225.

**Comment by itmo** [comment]
yes, trivially found using findbugs, in addition to 54 other unreported errors.. =D

**Comment by d2s** [comment]
Might also make sense to test the code with [Infer static analyzer](http://fbinfer.com/) (that is developed by Facebook for finding null pointer exceptions and resource leaks in Java code).

**Comment by kakoni** [comment]
Null pointer dereferences reported by infer
Found 12 issues

- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/Kasaaja.java#L327
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/Kasaaja.java#L862
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/ReseptiPurkaja.java#L163
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/ReseptiPurkaja.java#L166
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/ReseptiPurkaja.java#L171
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/ReseptiPurkaja.java#L156
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/ReseptiPurkaja.java#L160
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/ReseptinKorjausKasaaja.java#L430
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/ReseptinKorjausKasaaja.java#L430
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/ReseptinLukituksenPurkuKasaaja.java#L76
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/ReseptinMitatointiKasaaja.java#L269
- https://github.com/TheFinnishSocialInsuranceInstitution/KantaCDA-API/blob/master/KantaCDA-API/src/main/java/fi/kela/kanta/cda/ReseptinUusimisKasaaja.java#L359


Fixable by checking for null. Should I PR or will this go into Kela backlog?
